src/test/ui/consts/const-eval/ub-uninhabit.rs

```diff
@@ -16,13 +16,13 @@ use std::mem;
 enum Bar {}
 
 const BAD_BAD_BAD: Bar = unsafe { mem::transmute(()) };
-//~^ ERROR this constant likely exhibits undefined behavior
+//~^ ERROR this constant cannot be used
```

Comment:
You could try using an `union`-based transmute instead of `mem::transmute` to keep the old error. That would then keep testing the codepaths this is intended to test.